File: app/src/main/java/com/example/foofgeek/category_adapter.java
package com.example.foofgeek;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.foofgeek.Helper.ImageHelper;
import com.example.foofgeek.Model.CategoryItem;
import java.util.ArrayList;


public class category_adapter extends BaseAdapter {

    private LayoutInflater inflater;
    private ArrayList<CategoryItem> categoryList;
    private int layout;
    private ImageHelper imageHelper = new ImageHelper();

    public category_adapter(Context context, ArrayList<CategoryItem> categoryList, int layout) {
        this.inflater=(LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        this.categoryList = categoryList;
        this.layout = layout;
    }

    @Override
    public int getCount() {
        return categoryList.size();
    }

    @Override
    public Object getItem(int position) {
        return categoryList.get(position).get_category();
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if(convertView==null){
            convertView=inflater.inflate(layout,parent,false);
        }
        CategoryItem categoryItem = categoryList.get(position);
        ImageView icon= convertView.findViewById(R.id.categoryitem_img);
        icon.setImageBitmap(imageHelper.getBitmapFromByteArray(categoryItem.get_mainImg()));

        TextView name = convertView.findViewById(R.id.categoryitem_text);
        name.setText(categoryItem.get_category());
        return convertView;
    }
}

File: FoodGeek1/app/src/main/java/com/example/foofgeek/category_adapter.kt
package com.example.foofgeek

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.example.foofgeek.Helper.ImageHelper
import com.example.foofgeek.Model.CategoryItem

class category_adapter(context: Context?, categoryList: ArrayList<CategoryItem>, layout: Int) :
    BaseAdapter() {
    private val inflater: LayoutInflater
    private val categoryList: ArrayList<CategoryItem?>?
    private val layout: Int
    private val imageHelper = ImageHelper()
    private val name: TextView
        get() {
            TODO()
        }

    init {
        inflater = context!!.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        this.categoryList = categoryList
        this.layout = layout
    }

    override fun getCount(): Int {
        return categoryList!!.size
    }

    override fun getItem(position: Int): Any {
        return categoryList!![position]!!.get_category()
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View, parent: ViewGroup): View {
        var convertView = convertView
        if (convertView == null) {
            convertView = inflater.inflate(layout, parent, false)
        }
        val categoryItem = categoryList!![position]
        val icon = convertView.findViewById<ImageView>(R.id.categoryitem_img)
        val name = convertView.findViewById<TextView>(R.id.categoryitem_text)
        name.text = categoryItem!!.get_category()

        icon.setImageBitmap(imageHelper.getBitmapFromByteArray(categoryItem!!.get_mainImg()))

        return convertView
    }
}